### DISCUSSION TITLE
Trying to Upgrade knowlg on AWS from 5.7 to 6.1.0 facing issues

### DISCUSSION THREAD

**Discussion by AbhilashKD** [project-sunbird/sunbird-devops#4023] [Q&A]
Hi,
We are trying to upgrade the knowlg on AWS. Earlier we successfully deployed the knowlg 5.2 on AWS using helm charts but while upgrading the knowlg to 6.1.0 we are facing issues in Flink jobs and neo4j logstash the more details as follows : 

for upgradation we are referring to a repo : **https://github.com/Sunbird-Knowlg/knowledge-platform** with branch release-6.1.0

Changes made : 
1. Changed the cloud platform from Azure to AWS 
2. Changed the Azure blob storage with AWS s3
3. Added variables for AWS to access the s3 and other resources
4. Added variable for AWS region as well

after doing changes in the helm charts of the Knowlg-automation in above mentioned repo. trying to deploy the knowlg on AWS 
So except flink job for task manager and job manager and logstash under neo4j deployment facing issues. 
The issues and snapshots are attached below : 
I have did the changes in  helm charts and given the deployment so below are the errors iam getting :
in flink jobs :
`Flink Prod Job manager : 
```
Type     Reason       Age                  From     Message                                                                                                                                           Warning  FailedMount  22s (x746 over 25h)  kubelet  MountVolume.SetUp failed for volume "flink-config-volume" : configmap references non-existent config key: flink-conf
```

`Flink Prod Task manager :
```
Type     Reason       Age                  From     Message                                                                                                                                 Warning  FailedMount  97s (x746 over 25h)  kubelet  MountVolume.SetUp failed for volume "flink-config-volume" : configmap references non-existent config key: flink-conf  
[12:59](https://tekditech.slack.com/archives/C06FED4TPA8/p1733470179450029?thread_ts=1733462521.731339&cid=C06FED4TPA8)
```

and in Neo4j logstash facing error as :
 ```
[2024-12-06T07:25:10,532][WARN ][org.apache.kafka.clients.ClientUtils] Couldn't resolve server kafka-headless.sunbird.svc.cluster.local:9092 from bootstrap.servers as DNS resolution failed for kafka-he │
│ [2024-12-06T07:25:10,534][INFO ][org.apache.kafka.clients.producer.KafkaProducer] [Producer clientId=producer-1] Closing the Kafka producer with timeoutMillis = 0 ms.                                    │
│ [2024-12-06T07:25:10,541][ERROR][logstash.outputs.kafka   ] Unable to create Kafka producer from given configuration {:kafka_error_message=>org.apache.kafka.common.KafkaException: Failed to construct k │
│ [2024-12-06T07:25:10,562][ERROR][logstash.pipeline        ] Error registering plugin {:pipeline_id=>"main", :plugin=>"#<LogStash::OutputDelegator:0x5c200f0b>", :error=>"Failed to construct kafka produc │
│ [2024-12-06T07:25:10,568][ERROR][logstash.pipeline        ] Pipeline aborted due to error {:pipeline_id=>"main", :exception=>org.apache.kafka.common.KafkaException: Failed to construct kafka producer,  │
│ [2024-12-06T07:25:10,589][ERROR][logstash.agent           ] Failed to execute action {:id=>:main, :action_type=>LogStash::ConvergeResult::FailedAction, :message=>"Could not execute action: PipelineActi │
│ [2024-12-06T07:25:11,022][INFO ][logstash.agent           ] Successfully start
[sunbird-inquirybb-cassandra-migration-job-psb7r-1734004908027614895.log](https://github.com/user-attachments/files/18111101/sunbird-inquirybb-cassandra-migration-job-psb7r-1734004908027614895.log)
ed Logstash API endpoint {:port=>9600}                                                                                      │
[sunbird-neo4j-568cbb8685-ttkfg-1734004775773841496.log](https://github.com/user-attachments/files/18111076/sunbird-neo4j-568cbb8685-ttkfg-1734004775773841496.log)

│ [2024-12-06T07:25:15,775][INFO ][logstash.runner          ] Logstash shut down.                                                                                                                           │
│ [2024-12-06T07:25:15,821][ERROR][org.logstash.Logst
[sunbird-neo4j-568cbb8685-ttkfg-1734004775773841496.log](https://github.com/user-
![All-services](https://github.com/user-attachments/assets/b4595e95-e626-4d19-b9fd-43cadb781b8c)
![inquiry-error](https://github.com/user-attachments/assets/8cc66ecc-7110-4f08-bcd0-4996621c9dc0)
attachments/files/18111084/sunbird-neo4j-568cbb8685-ttkfg-1734004775773841496.log)
[sunbird-neo4j-568cbb8685-ttkfg-1734004775773841496.log](https://github.com/user-attachments/files/18111092/sunbird-neo4j-568cbb8685-ttkfg-1734004775773841496.log)
ash    ] java.lang.IllegalStateException: Logstash stopped processing because of an error: (SystemExit) exit
```
@pallakartheekreddy @gandham-santhosh @aimansharief @rhwarrier @coolbung @snehal09004 @rajnishdargan
![All-services](https://github.com/user-attachments/assets/8efcc848-8261-4b72-a8da-043ef6cb673b)
![inquiry-error](https://github.com/user-attachments/assets/98a20f62-8134-4376-aef6-6b84ec9351bd)
[pods-all-1732167530473705762.csv](https://github.com/user-attachments/files/18111283/pods-all-1732167530473705762.csv)
[sunbird-inquirybb-cassandra-migration-job-psb7r-1734004908027614895.log](https://github.com/user-attachments/files/18111286/sunbird-inquirybb-cassandra-migration-job-psb7r-1734004908027614895.log)
[sunbird-neo4j-568cbb8685-ttkfg-1734004775773841496.log](https://github.com/user-attachments/files/18111288/sunbird-neo4j-568cbb8685-ttkfg-1734004775773841496.log)

**Comment by aimansharief** [comment]
@AbhilashKD The Flink error appears to be caused by an issue in the ConfigMap, while the errors in the Logstash and Cassandra jobs seem to be related to connection problems with their respective databases.

**Comment by aimansharief** [comment]
We can connect tomorrow to resolve the issue

**Comment by AbhilashKD** [comment]
Yes sure

**Reply by AbhilashKD** [reply]
@aimansharief at what time you're available today to connect so will look into this ?

**Reply by aimansharief** [reply]
can you send an invite to connect at 2'45

**Reply by AbhilashKD** [reply]
@aimansharief 
Shared invite with you

**Reply by AbhilashKD** [reply]
@aimansharief 
Waiting for you on call

**Reply by aimansharief** [reply]
@AbhilashKD Sorry, previous call got extended . I'm joining now

**Reply by aimansharief** [reply]
Please find the link to the docker file here:
https://github.com/Sunbird-Knowlg/knowledge-platform/blob/release-6.1.0/build/content-service/Dockerfile

Steps to build and push docker image to your repo
```
docker build -f build/content-service/Dockerfile -t <repository>/content-service:6.0.1 .
docker tag <repository>/content-service:6.0.1
docker push <repository>/content-service:6.0.1
```

Also follow the below steps before building docker image
https://github.com/Sunbird-Knowlg/knowledge-platform/tree/release-6.1.0?tab=readme-ov-file#running-content-service

**Comment by aimansharief** [comment]
Once you fix the kafka issue, this is where you configure kafka-logstash connection url.
https://github.com/Sunbird-Knowlg/knowlg-automation/blob/c481a8dbca2a7a038132c1d1947b5476ac62f7ad/terraform/modules/helm/neo4j/neo4j-helm-chart/values.yaml#L18

Please update it to fix the existing error.

**Comment by AbhilashKD** [comment]
@aimansharief 
Can you share me the images directly so that i can take a pull and check all the configs by deploying it to my EKS cluster.
As the building images its a long process also if we catch in an issue then it will again take more time to debug it or you also need to assist us while debugging and troubleshooting.
So if you share images directly with 6.1.0 then it will be helpful for us.
@coolbung @gandham-santhosh @rhwarrier @pallakartheekreddy  @snehal0904

**Reply by aimansharief** [reply]
@AbhilashKD Please find the list of images for knowlg-6.1.0 :

```
sunbirded.azurecr.io/content-service:release-6.1.0_RC5_f6fb77d_893
sunbirded.azurecr.io/taxonomy-service:release-6.1.0_RC4_2272e93_47
sunbirded.azurecr.io/search-service:release-6.1.0_RC4_2272e93_40
sunbirded.azurecr.io/knowledge-platform-jobs:release-6.1.0_RC1_5470dbe_87
sunbirded.azurecr.io/sunbird-dial-service:release-6.1.0_RC2_213843a_30
```
Let me know if you need any other images.

**Reply by AbhilashKD** [reply]
@aimansharief 
Thanks for sharing this.
Can you create and share images for Assessment and Learn also we need that in our current flow.

**Reply by AbhilashKD** [reply]
@aimansharief any updates on above requirement ?
Also please share Image for the Assessment service is upgraded to release-8.0.0 we need this to check ongoing workflow.

**Reply by aimansharief** [reply]
@AbhilashKD Below are the images related to Lern
```
sunbirded.azurecr.io/adminutil:release-4.5.0_RC3_68b1eaa_61
sunbirded.azurecr.io/cert_service:release-5.0.1_RC2
sunbirded.azurecr.io/sunbird-rc-certificate-api:release-5.1.0_2bc7760_10
sunbirded.azurecr.io/sunbird-rc-certificate-signer:release-5.1.0_2bc7760_9
sunbirded.azurecr.io/cert_registry_service:release-5.0.0_RC1_1
sunbirded.azurecr.io/data-pipeline:release-7.0.0_RC5_4a07143_21
sunbirded.azurecr.io/groups_service:release-7.0.0_RC1_e82e95d_3
sunbirded.azurecr.io/lms_service:release-7.0.0_RC2_d14e96b_32
sunbirded.azurecr.io/notification_service:release-7.0.0_RC2_42c4145_8
sunbirded.azurecr.io/print-service:release-3.0.0_RC1
sunbirded.azurecr.io/sunbird-rc-core:release-5.1.0_ece7ed3_14
sunbirded.azurecr.io/userorg-service:release-7.0.0_RC5_0d513be_31
```

**Reply by AbhilashKD** [reply]
@aimansharief 
Thanks for the help. 
We need assessment Images as well 
for release 8 also 
This is important for checking our flow 
Please do the needful for this

**Reply by aimansharief** [reply]
@AbhilashKD 
`sunbirded.azurecr.io/assessment-service:release-7.0.0_RC2_aeccd6e_16`

**Reply by AbhilashKD** [reply]
@aimansharief 
We need release 8 as well 
We are doing two tasks simultaneously for that we need release 8 of assessment as well

**Reply by AbhilashKD** [reply]
@aimansharief can you please share 
need this images on priority for our testing the overall flow

**Comment by rajnishdargan** [comment]
Hi @aimansharief,

I hope you’re doing well. 
In our current setup of **Knowlg** and **inQuiry**, I noticed the following versions are being used:

- **Content Service:** sunbirded.azurecr.io/content-service:release-6.1.0_RC5_f6fb77d_893
- **Content Publish Job Manager:** sunbirded.azurecr.io/knowledge-platform-jobs:release-5.5.0_RC2_037e2e9_84
- **Content Publish Task Manager:** sunbirded.azurecr.io/knowledge-platform-jobs:release-5.5.0_RC2_037e2e9_84

Could you please confirm if this setup is correct? Should we upgrade the **Content Publish Job Manager** and **Task Manager** to match the Content Service version?

Additionally, could you clarify:

Are there any **other services/jobs** that require an upgrade?
Should we make any changes to the **config map** or **schema** as part of these upgrades? If yes, where can we find the relevant details, including the compatible schema and configuration?

**Setup Details**
We are working on a fresh setup using the [Ed installer](https://github.com/project-sunbird/sunbird-ed-installer) using release-7.0.0 branch. Since there is no release-8.0.0 branch, we are modifying the Ed installer to:

Install **version 6.1** for Knowlg service.
Install **version 8.0** for inQuiry service.
Before proceeding with configuring the channel, framework, object category, and definition, I need confirmation that this setup is fine.

**Assessment Service Upgrade**
Lastly, we also need to upgrade the Assessment Service to version 8.0.0. Could you please share the steps and image details required for this upgrade?

Let me know if you need any additional information to confirm the setup.
@pallakartheekreddy @rhwarrier @coolbung @snehal0904 @AbhilashKD

**Comment by AbhilashKD** [comment]
@aimansharief 
Can you please share me the sunbirded.azurecr.io/knowledge-platform-jobs image for the release-5.1 ??
As we are doing one exercise for that i need knowledge-platform job image version as 5.1

**Reply by AbhilashKD** [reply]
@aimansharief 
Any Updates On this ??

**Reply by aimansharief** [reply]
@AbhilashKD I'll have to build new image for this, will try to build and let you know.

**Reply by AbhilashKD** [reply]
ok thanks @aimansharief 
how much time i can expect for this ?
